torchvision/transforms/functional.py

```diff
@@ -431,7 +431,9 @@ def resize(
     return F_t.resize(img, size=size, interpolation=interpolation.value, max_size=max_size, antialias=antialias)
 
 
-def pad(img: Tensor, padding: List[int], fill: int = 0, padding_mode: str = "constant") -> Tensor:
+def pad(
+    img: Tensor, padding: List[int], fill: Union[List[float], float] = 0.0, padding_mode: str = "constant"
```

Comment:
The default value changed from int to float. JIT will fail if we pass integers.

Reply:
The `int` used previously is very misleading. We mainly use floats because our tensors get rescaled as you know. Unfortunately adding both `List[int]` and `List[float]` in the union doesn't work due to JIT issues. See https://github.com/pytorch/pytorch/issues/69434